code/modules/mob/living/simple_animal/friendly/farm/traits/other.dm

```diff
@@ -0,0 +1,127 @@
+/datum/farm_animal_trait/talkative
+	name = "Talkative"
+	description = "This animal will attempt to talk more often and mimic what others say."
+	manifest_probability = 55
+	continue_probability = 75
+
+/datum/farm_animal_trait/talkative/on_apply(var/mob/living/simple_animal/farm/M)
+	M.speak_chance = 15
+	return
+
+/datum/farm_animal_trait/talkative/on_hear(message, atom/movable/speaker, message_langs, raw_message, radio_freq, list/spans)
+	if(speaker != owner.owner && prob(40)) //Dont imitate ourselves
+		if(owner.owner.speak.len >= 40)
+			owner.owner.speak -= pick(owner.owner.speak)
+		owner.owner.speak |= html_decode(raw_message)
+
+/datum/farm_animal_trait/bioluminescent
+	name = "Bioluminescent"
+	description = "This animal is bioluminescent, and glows."
+	manifest_probability = 55
+	continue_probability = 75
+
+/datum/farm_animal_trait/bioluminescent/on_apply(var/mob/living/simple_animal/farm/M)
+	M.SetLuminosity(4)
+	return
+
+/datum/farm_animal_trait/spooky
+	name = "Spooky"
+	description = "This animal is transparent and spooky."
+	manifest_probability = 55
+	continue_probability = 75
+
+/datum/farm_animal_trait/spooky/on_apply(var/mob/living/simple_animal/farm/M)
+	animate(M, alpha = 127, time = 30)
+	return
+
+/datum/farm_animal_trait/spooky/on_death(var/mob/living/simple_animal/farm/M)
+	M.visible_message("[M] fades away...")
+	animate(M, alpha = 0, time = 30)
+	spawn(30)
```

Comment:
Im pretty sure theres an argument in animate() proc that makes animation either poarallel to other computation or halts everything until its finished